runtime-transaction/src/runtime_transaction/transaction_view.rs

```diff
@@ -91,6 +100,47 @@ impl<D: TransactionData> RuntimeTransaction<ResolvedTransactionView<D>> {
     }
 }
 
+impl<D: TransactionData> TransactionWithMeta for RuntimeTransaction<ResolvedTransactionView<D>> {
+    fn as_sanitized_transaction(&self) -> impl Borrow<SanitizedTransaction> {
+        let VersionedTransaction {
+            signatures,
+            message,
+        } = self.to_versioned_transaction();
+
+        let is_writable_account_cache = (0..self.transaction.total_num_accounts())
+            .map(|index| self.is_writable(usize::from(index)))
+            .collect();
```

Comment:
How about directly cloning the writable cache that's already stored in the resolved view?

Reply:
I'm planning to evaluate using a bitset to avoid allocating here

Reply:
then the only allocation will be for ALT using transactions, still not great but not as bad as it once was 😄